src/main/java/org/graylog/integrations/dataadapters/GreyNoiseCommunityIpLookupAdapter.java

```diff
@@ -125,27 +136,53 @@ protected LookupResult doGet(Object ipAddress) {
         return result;
     }
 
-    private Request createRequest(Object ipAddress) {
-        String ipString = ipAddress == null? "" : ipAddress.toString();
-        if(ipString.trim().isEmpty()){
-            String error = String.format("[%s] requires an IP address to perform Lookup",ADAPTER_NAME);
-            throw new IllegalArgumentException(error);
-        }
+    @VisibleForTesting
+    Optional<Request> createRequest(Object ipAddress) {
+        Optional<String> ipString = getValidIpString(ipAddress);
 
-        String userAgent = String.format(USER_AGENT, Version.CURRENT_CLASSPATH);
+        if (!ipString.isPresent()) {
+            return Optional.empty();
+        }
         String apiToken = encryptedValueService.decrypt(config.apiToken());
-        if(apiToken == null || apiToken.trim().isEmpty()){
-            String error = String.format("[%s] requires a non-null API Token",ADAPTER_NAME);
+        if (apiToken == null || apiToken.trim().isEmpty()) {
+            String error = String.format("[%s] requires a non-null API Token", ADAPTER_NAME);
             throw new IllegalArgumentException(error);
         }
 
-        return new Request.Builder()
-                .url(String.join("/", GREYNOISE_COMMUNITY_ENDPOINT, ipString))
+        Request request = new Request.Builder()
+                .url(String.join("/", GREYNOISE_COMMUNITY_ENDPOINT, ipString.get()))
                 .method(METHOD, null)
                 .addHeader("Accept", ACCEPT_TYPE)
-                .addHeader("User-Agent", userAgent)
+                .addHeader("User-Agent", USER_AGENT)
                 .addHeader("key", apiToken)
                 .build();
+
+        return Optional.of(request);
+    }
+
+    private Optional<String> getValidIpString(Object ipAddress) {
+        final String ipString = ipAddress == null ? "" : ipAddress.toString();
+        if (ipString.trim().isEmpty()) {
+            String error = String.format("'%s' requires an IP address to perform Lookup", ADAPTER_NAME);
+            throw new IllegalArgumentException(error);
+        }
+
+        final String validIpAddress;
+        final InetAddressValidator validator = InetAddressValidator.getInstance();
+        if (validator.isValidInet6Address(ipString)) {
+            LOG.error("'{}' is an IPv6 Address.  '{}' does not support IPv6 Addresses", ipAddress, ADAPTER_NAME);
```

Comment:
Do you think Reserved and IPv6 addresses may be better off being logged at a lesser severity - `WARN` or possibly even `INFO`? There is nothing wrong with those IPs coming through Graylog we just aren't going to check them with GreyNoise because there is no need. Spitting out an error log message for that seems excessively severe. 

Invalid IPv4 can still be logged at the error level since that would definitely be a problem. We may also want to look at possibly including an error message in the lookup if that does happen. We should be able to merge it without it, but it may be beneficial to give some information about why there was a lookup error in the result itself if that is possible... I haven't worked with look up tables enough to know if we handle something like that already.

Reply:
I see what you mean.  My point of view was that, from the adapter itself, IPv6 is an invalid input-error.  @danotorrey , do you have any thoughts? Is this better off as a **Warning**?

I discussed the message with @waab76 .  What (I understood) he said is that the message is not really useful in a lookup table as the user has no good way of seeing it--the user only cares whether there is a value.